File: JAVA/Assignment 12/Program1.java
/////////////////////////////////////////////////////////////////////////////////////////
//                                                                                     //
// Problem Statement : Write a Java program to find maximum difference between two     //
//                     elements such that larger element appears after smaller number. //
// Author : Pallavi Omprakash Malewar                                                  //
// Date : 4 Aug 2026                                                                   //
//                                                                                     //
/////////////////////////////////////////////////////////////////////////////////////////


/*  

    Input : array : [2, 3, 10, 6, 4, 8, 1]

    Output: 8 (10 - 2)

    Input : array : [7, 9, 5, 6, 3, 2]

    Output: 2 (7 - 9)

*/


import java.util.Scanner;

class MaximumDifference
{
    public int MaxDifference(int Arr[], int size)
    {
        int i = 0, j = 0;
        int Max = 0;
        for(i = 0 ; i < size ; i++)
        {
            for(j = i + 1 ; j < size ; j++)
            {
                int Diff = Arr[j] - Arr[i];
                if((Max < Diff) && (i < j)) Max = Diff;
            }
        }
        return Max;
    }
}

class Program1
{
    public static void main(String A[])
    {
        MaximumDifference mobj = new MaximumDifference();

        int arr[] = {7, 9, 5, 6, 3, 2};

        System.out.println("Maximum Difference is : " + mobj.MaxDifference(arr, arr.length));
    }
}

File: CPP/Assignment8/Program1.cpp
////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
// File name :   Program1.cpp
// Description : Write a program which accept matrix from user & return addition of diagonal elements.
// Author :      Pallavi Omprakash Malewar
// Date :        27/07/2026
//
////////////////////////////////////////////////////////////////////////////////////////////////////////////

/*

    Input  :    3   2   5   9
                4   3   2   2
                8   4   1   5   
                3   9   7   5

    Output :    12

*/


#include<iostream>
using namespace std;

int AddDiagonal(int Arr[][4], int iRow, int iCol)
{
    int iSum = 0;

    for(int i = 0 ; i < iRow ; i++)
    {
        iSum += Arr[i][i];
    }
    return iSum;
}

int main()
{
    int Arr[4][4]; 
    int iRet = 0;

    cout << "Enter the elements :\n ";

    for(int i = 0 ; i < 4 ; i++)
    {
        for(int j = 0 ; j < 4 ; j++)
        {
            cin >> Arr[i][j];
        }
        cout << endl;
    }
    iRet = AddDiagonal(Arr, 4, 4);
    cout << "Addition of Diagonal : " << iRet << endl;

    return 0;
}